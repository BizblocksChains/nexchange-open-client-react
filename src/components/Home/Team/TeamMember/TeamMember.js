import React from 'react';
import styles from './TeamMember.scss';

const TeamMember = props => {
  const social = [];
  Object.keys(props.social).forEach(key => {
    let className = `fab fa-${key}`;
    if (key === 'rss') {
      className = 'fas fa-rss';
    } else if (key === 'linkedin') {
      className = `fab fa-linkedin-in`;
    }

    social.push(
      <a href={props.social[key]} className={styles[key]} key={key} target="_blank">
        <i className={className} />
      </a>
    );
  });

  return (
    <div className="col-xs-12 col-ms-6 col-sm-4">
      <div className={styles.member}>
        <div className={styles.profile}>
          <img src={require(`Img/team/${props.id}.jpg`)} alt={`${props.name} profile`} />
        </div>

        <h3 className={styles.name}>{props.name}</h3>
        <div className={styles.social}>{social}</div>
        <div className={styles.description}>{props.description}</div>
      </div>
    </div>
  );
};

export default TeamMember;
