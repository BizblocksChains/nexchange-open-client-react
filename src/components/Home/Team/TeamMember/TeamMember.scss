.member {
  margin-bottom: 75px;
  text-align: center;

  @media (min-width: $screen-sm-min) {
    padding-right: 15%;
  }
}

.profile {
  display: inline-block;
  position: relative;
  width: 200px;
  max-width: 225px;

  @media (min-width: $screen-md-min) {
    width: 220px;
  }

  img {
    width: 100%;
    filter: grayscale(100%);
  }
}

.social {
  margin: 10px 0;

  a {
    display: inline-block;
    height: 42px;
    width: 42px;
    background: #efefef;
    border-radius: 100px;
    color: #9c9c9c;
    text-align: center;
    padding-top: 7px;
    margin: 0 5px;
    font-size: 20px;

    &:hover {
      &.wikipedia-w {
        background: #494f57;
      }
      &.twitter {
        background: #1da1f2;
      }
      &.medium {
        background: #3cb371;
      }
      &.linkedin {
        background: #0077b5;
      }
      &.github {
        background: #222222;
      }
      &.rss {
        background: #f26522;
      }
      &.stack-overflow {
        background: #f48024;
      }
      &.facebook {
        background: #3b5998;
      }
      &.slack {
        background: #e01563;
      }
      &.telegram {
        background: #0088cc;
      }

      color: #fff;
    }
  }
}

.name {
  margin: 15px 0;
  font-size: 18px;
  font-weight: bold;
  color: $black;
}

.description {
  font-size: 14px;
  color: #9c9c9c;

  p {
    font-size: 14px;
  }

  a {
    text-decoration: underline;
    font-weight: 700;
  }
}
