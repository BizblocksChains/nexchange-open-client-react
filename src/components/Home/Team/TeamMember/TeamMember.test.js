import React from 'react';
import { shallow } from 'enzyme';
import TeamMember from './TeamMember';

describe('TeamMember', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(
      <TeamMember
        id="vit"
        name="Vít Jedlička"
        description="President of the Free Republic of Liberland. Former Czech politician, publicist and activist."
        social={{
          linkedin: 'https://www.linkedin.com/in/vit-jedlicka-9115aa2/',
          rss: 'http://www.google.com',
          blog: 'https://www.linkedin.com/in/vit-jedlicka-9115aa2/',
        }}
      />
    );
    expect(wrapShallow).toMatchSnapshot();
  });
});
