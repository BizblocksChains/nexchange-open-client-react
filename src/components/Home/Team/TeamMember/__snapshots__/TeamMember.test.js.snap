// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMember renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TeamMember
    description="President of the Free Republic of Liberland. Former Czech politician, publicist and activist."
    id="vit"
    name="Vít Jedlička"
    social={
        Object {
            "blog": "https://www.linkedin.com/in/vit-jedlicka-9115aa2/",
            "linkedin": "https://www.linkedin.com/in/vit-jedlicka-9115aa2/",
            "rss": "http://www.google.com",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="member"
>
        <div
                className="profile"
        >
                <img
                        alt="Vít Jedlička profile"
                        src="vit.jpg"
                />
        </div>
        <h3
                className="name"
        >
                Vít Jedlička
        </h3>
        <div
                className="social"
        >
                <a
                        className="linkedin"
                        href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                        target="_blank"
                >
                        <i
                                className="fab fa-linkedin-in"
                        />
                </a>
                <a
                        className="rss"
                        href="http://www.google.com"
                        target="_blank"
                >
                        <i
                                className="fas fa-rss"
                        />
                </a>
                <a
                        className="blog"
                        href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                        target="_blank"
                >
                        <i
                                className="fab fa-blog"
                        />
                </a>
        </div>
        <div
                className="description"
        >
                President of the Free Republic of Liberland. Former Czech politician, publicist and activist.
        </div>
</div>,
      "className": "col-xs-12 col-ms-6 col-sm-4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="profile"
>
            <img
                        alt="Vít Jedlička profile"
                        src="vit.jpg"
            />
</div>,
          <h3
            className="name"
>
            Vít Jedlička
</h3>,
          <div
            className="social"
>
            <a
                        className="linkedin"
                        href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                        target="_blank"
            >
                        <i
                                    className="fab fa-linkedin-in"
                        />
            </a>
            <a
                        className="rss"
                        href="http://www.google.com"
                        target="_blank"
            >
                        <i
                                    className="fas fa-rss"
                        />
            </a>
            <a
                        className="blog"
                        href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                        target="_blank"
            >
                        <i
                                    className="fab fa-blog"
                        />
            </a>
</div>,
          <div
            className="description"
>
            President of the Free Republic of Liberland. Former Czech politician, publicist and activist.
</div>,
        ],
        "className": "member",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Vít Jedlička profile"
              src="vit.jpg"
/>,
            "className": "profile",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Vít Jedlička profile",
              "src": "vit.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vít Jedlička",
            "className": "name",
          },
          "ref": null,
          "rendered": "Vít Jedlička",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="linkedin"
                href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                target="_blank"
>
                <i
                                className="fab fa-linkedin-in"
                />
</a>,
              <a
                className="rss"
                href="http://www.google.com"
                target="_blank"
>
                <i
                                className="fas fa-rss"
                />
</a>,
              <a
                className="blog"
                href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                target="_blank"
>
                <i
                                className="fab fa-blog"
                />
</a>,
            ],
            "className": "social",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "linkedin",
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fab fa-linkedin-in"
/>,
                "className": "linkedin",
                "href": "https://www.linkedin.com/in/vit-jedlicka-9115aa2/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fab fa-linkedin-in",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": "rss",
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-rss"
/>,
                "className": "rss",
                "href": "http://www.google.com",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-rss",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": "blog",
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fab fa-blog"
/>,
                "className": "blog",
                "href": "https://www.linkedin.com/in/vit-jedlicka-9115aa2/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fab fa-blog",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "President of the Free Republic of Liberland. Former Czech politician, publicist and activist.",
            "className": "description",
          },
          "ref": null,
          "rendered": "President of the Free Republic of Liberland. Former Czech politician, publicist and activist.",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="member"
>
          <div
                    className="profile"
          >
                    <img
                              alt="Vít Jedlička profile"
                              src="vit.jpg"
                    />
          </div>
          <h3
                    className="name"
          >
                    Vít Jedlička
          </h3>
          <div
                    className="social"
          >
                    <a
                              className="linkedin"
                              href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                              target="_blank"
                    >
                              <i
                                        className="fab fa-linkedin-in"
                              />
                    </a>
                    <a
                              className="rss"
                              href="http://www.google.com"
                              target="_blank"
                    >
                              <i
                                        className="fas fa-rss"
                              />
                    </a>
                    <a
                              className="blog"
                              href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                              target="_blank"
                    >
                              <i
                                        className="fab fa-blog"
                              />
                    </a>
          </div>
          <div
                    className="description"
          >
                    President of the Free Republic of Liberland. Former Czech politician, publicist and activist.
          </div>
</div>,
        "className": "col-xs-12 col-ms-6 col-sm-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="profile"
>
              <img
                            alt="Vít Jedlička profile"
                            src="vit.jpg"
              />
</div>,
            <h3
              className="name"
>
              Vít Jedlička
</h3>,
            <div
              className="social"
>
              <a
                            className="linkedin"
                            href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                            target="_blank"
              >
                            <i
                                          className="fab fa-linkedin-in"
                            />
              </a>
              <a
                            className="rss"
                            href="http://www.google.com"
                            target="_blank"
              >
                            <i
                                          className="fas fa-rss"
                            />
              </a>
              <a
                            className="blog"
                            href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                            target="_blank"
              >
                            <i
                                          className="fab fa-blog"
                            />
              </a>
</div>,
            <div
              className="description"
>
              President of the Free Republic of Liberland. Former Czech politician, publicist and activist.
</div>,
          ],
          "className": "member",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Vít Jedlička profile"
                src="vit.jpg"
/>,
              "className": "profile",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Vít Jedlička profile",
                "src": "vit.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vít Jedlička",
              "className": "name",
            },
            "ref": null,
            "rendered": "Vít Jedlička",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="linkedin"
                  href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                  target="_blank"
>
                  <i
                                    className="fab fa-linkedin-in"
                  />
</a>,
                <a
                  className="rss"
                  href="http://www.google.com"
                  target="_blank"
>
                  <i
                                    className="fas fa-rss"
                  />
</a>,
                <a
                  className="blog"
                  href="https://www.linkedin.com/in/vit-jedlicka-9115aa2/"
                  target="_blank"
>
                  <i
                                    className="fab fa-blog"
                  />
</a>,
              ],
              "className": "social",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "linkedin",
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fab fa-linkedin-in"
/>,
                  "className": "linkedin",
                  "href": "https://www.linkedin.com/in/vit-jedlicka-9115aa2/",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fab fa-linkedin-in",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": "rss",
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fas fa-rss"
/>,
                  "className": "rss",
                  "href": "http://www.google.com",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-rss",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": "blog",
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fab fa-blog"
/>,
                  "className": "blog",
                  "href": "https://www.linkedin.com/in/vit-jedlicka-9115aa2/",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fab fa-blog",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "President of the Free Republic of Liberland. Former Czech politician, publicist and activist.",
              "className": "description",
            },
            "ref": null,
            "rendered": "President of the Free Republic of Liberland. Former Czech politician, publicist and activist.",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
